stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt

```diff
@@ -5,92 +5,64 @@ import androidx.lifecycle.asLiveData
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.filterNotNull
-import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
-import kotlinx.coroutines.launch
+import io.getstream.chat.android.offline.controller.QueryChannelsController as QueryChannelsControllerStateFlow
 
 private const val MESSAGE_LIMIT = 10
 private const val MEMBER_LIMIT = 30
-private const val INITIAL_CHANNEL_OFFSET = 0
 private const val CHANNEL_LIMIT = 30
 
 internal class QueryChannelsControllerImpl(
-    override val filter: FilterObject,
-    override val sort: QuerySort<Channel>,
-    private val client: ChatClient,
-    private val domainImpl: ChatDomainImpl,
+    filter: FilterObject,
+    sort: QuerySort<Channel>,
+    client: ChatClient,
+    domainImpl: ChatDomainImpl,
 ) : QueryChannelsController {
-    override var newChannelEventFilter: (Channel, FilterObject) -> Boolean = { _, _ -> true }
-    override var recoveryNeeded: Boolean = false
+    private val controllerStateFlow =
+        QueryChannelsControllerStateFlow(filter, sort, client, domainImpl)
 
-    val queryChannelsSpec: QueryChannelsSpec = QueryChannelsSpec(filter, sort)
+    override val filter: FilterObject
+        get() = controllerStateFlow.filter
 
-    private val _endOfChannels = MutableStateFlow(false)
-    override val endOfChannels: LiveData<Boolean> = _endOfChannels.asLiveData()
+    override val sort: QuerySort<Channel>
+        get() = controllerStateFlow.sort
 
-    private val _channels = MutableStateFlow<Map<String, Channel>>(emptyMap())
-
-    private val _sortedChannels = _channels.filterNotNull()
-        .map { it.values.sortedWith(sort.comparator) }.stateIn(domainImpl.scope, SharingStarted.Eagerly, emptyList())
+    override var newChannelEventFilter: (Channel, FilterObject) -> Boolean
+        get() = controllerStateFlow.newChannelEventFilter
+        set(value) {
+            controllerStateFlow.newChannelEventFilter = value
+        }
 
-    // Keep the channel list locally sorted
-    override var channels: LiveData<List<Channel>> = _sortedChannels.asLiveData()
+    override var recoveryNeeded: Boolean
+        get() = controllerStateFlow.recoveryNeeded
+        set(value) {
+            controllerStateFlow.recoveryNeeded = value
+        }
+    val queryChannelsSpec: QueryChannelsSpec = controllerStateFlow.queryChannelsSpec
 
-    private val logger = ChatLogger.get("ChatDomain QueryChannelsController")
+    override val endOfChannels: LiveData<Boolean> = controllerStateFlow.endOfChannels.asLiveData()
 
-    private val _loading = MutableStateFlow(false)
-    override val loading: LiveData<Boolean> = _loading.asLiveData()
+    // Keep the channel list locally sorted
+    override val channels: LiveData<List<Channel>>
+        get() = controllerStateFlow._sortedChannels.asLiveData()
```

Comment:
Why `controllerStateFlow` doesn't have `channels: Flow<List<Channel>>`? It should have the same interface but in terms of `Flow<*>`. If it was so you could do:
```kotlin
override val channels: LiveData<List<Channel>> = controllerStateFlow.channels.asLiveData()
```